client/src/components/ActivityPane/index.js

```diff
@@ -74,28 +74,33 @@ export default function ActivityPane(props) {
       })
       .catch((error) => console.log(error));
 
-    try {
-      // Checks if the current user is also the owner of the repo
-      if (authState.user.username) {
-        const repository = props.location.pathname
-          .split("/")[2]
-          .replace("_", "/");
-        const authToken =
-          Cookies.get("gh_private_repo_token") || Cookies.get("gh_login_token");
-        const ghRepoResponse = await axios({
-          method: "get",
-          url: `${githubApiDomain}/repos/${repository}`,
-          headers: {
-            accept: "application/json",
-            Authorization: `token ${authToken}`,
-          },
-        });
+    const checkIfUserIsOwner = async () => {
+      try {
+        // Checks if the current user is also the owner of the repo
+        if (authState.user.username) {
+          const repository = props.location.pathname
+            .split("/")[2]
+            .replace("_", "/");
+          const authToken =
+            Cookies.get("gh_private_repo_token") ||
+            Cookies.get("gh_login_token");
+          const ghRepoResponse = await axios({
+            method: "get",
+            url: `${githubApiDomain}/repos/${repository}`,
+            headers: {
+              accept: "application/json",
+              Authorization: `token ${authToken}`,
+            },
+          });
 
-        setIsRepoOwner(ghRepoResponse.data.permissions.admin);
+          setIsRepoOwner(ghRepoResponse.data.permissions.admin);
+        }
+      } catch (error) {
+        console.log(error);
       }
-    } catch (error) {
-      console.log(error);
     }
+    checkIfUserIsOwner();
+    // eslint-disable-next-line
```

Comment:
Instead of disabling lint we should add `webhookSubscriptions` and `authState` to the useEffect dependency array here.

Reply:
Adding these will affect the logic.
(We want this hook to run on channel switch. Adding the above dependencies will run it when webhookSubscriptions change)